Source/DafnyLanguageServer/Workspace/Notifications/VerificationDiagnosticsParams.cs

```diff
@@ -0,0 +1,599 @@
+            previousNotUnknown != LineVerificationStatus.VerifiedVerifying) {
+          previousNotUnknown = LineVerificationStatus.Nothing;
+        }
+        if (result[line] == LineVerificationStatus.Nothing) {
+          result[line] = previousNotUnknown;
+        } else {
+          previousNotUnknown = result[line];
+        }
+      }
+
+      var resolutionErrorRendered = 0;
+      foreach (var diagnostic in diagnostics) {
+        if (resolutionErrorRendered >= numberOfResolutionErrors) {
+          break;
+        }
+        if (diagnostic.Range.Start.Line >= 0 && diagnostic.Range.Start.Line < result.Length) {
+          result[diagnostic.Range.Start.Line] = LineVerificationStatus.ResolutionError;
+          resolutionErrorRendered++;
+        }
+      }
+
+      var existsErrorRange = false;
+      var existsError = false;
+      foreach (var line in result) {
+        existsErrorRange = existsErrorRange || line == LineVerificationStatus.ErrorContext;
+        existsError = existsError || line == LineVerificationStatus.AssertionFailed;
+      }
+
+      if (existsErrorRange && !existsError) {
+        existsError = false;
+      }
+
+      return result;
+    }
+  }
+
+
+  public enum VerificationStatus {
+    Nothing = 0,
+    Verified = 200,
+    Inconclusive = 270,
+    Error = 400
+  }
+
+  public enum CurrentStatus {
+    Current = 0,
+    Obsolete = 1,
+    Verifying = 2
+  }
+
+  public enum LineVerificationStatus {
+    // Default value for every line, before the renderer figures it out.
+    Nothing = 0,
+    // For first-time computation not actively computing but soon. Synonym of "obsolete"
+    // (scheduledComputation)
+    Scheduled = 1,
+    // For first-time computations, actively computing
+    Verifying = 2,
+    VerifiedObsolete = 201,
+    VerifiedVerifying = 202,
+    // Also applicable for empty spaces if they are not surrounded by errors.
+    Verified = 200,
+    // For trees containing children with errors (e.g. functions, methods, fields, subset types)
+    ErrorContextObsolete = 301,
+    ErrorContextVerifying = 302,
+    ErrorContext = 300,
+    // For individual assertions in error contexts
+    AssertionVerifiedInErrorContextObsolete = 351,
+    AssertionVerifiedInErrorContextVerifying = 352,
+    AssertionVerifiedInErrorContext = 350,
+    // For specific lines which have errors on it. They take over verified assertions
+    AssertionFailedObsolete = 401,
+    AssertionFailedVerifying = 402,
+    AssertionFailed = 400,
+    // For lines containing resolution or parse errors
+    ResolutionError = 500
+  }
+
+  /// <summary>
+  /// A verification tree is an abstraction over the code to represent the verification
+  /// status of a region of the document, useful for IDE verification inspection.
+  /// A verification tree can contain other child trees.
+  /// It can currently be rendered linearly, e.g. for gutter display, or used as a tree in a test-like display. 
```

Comment:
I'm unable to understand what this tree is doing. I feel like it's supposed to provide information for many different features. I'd like to have separate APIs for different features and to be able to verify that each API sends the minimal amount of required information, but with this huge record that also contains a lot of logic I find it impossible to verify that.